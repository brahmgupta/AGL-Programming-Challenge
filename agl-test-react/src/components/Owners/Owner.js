// @flow
import React from "react";
import Pet from "./Pet";

type OwnerType = {
  ownerGender: string,
  catNames: Array<string>
};

type PropsType = {
  owner: OwnerType
};

const Owner = ({ owner }: PropsType) => {
  return (
    <li data-testid="owner" className="owner">
      {owner.ownerGender}
      <ul className="pet-list">
        {owner.catNames &&
          owner.catNames.map(petName => (
            <Pet key={petName} petName={petName} />
          ))}
      </ul>
    </li>
  );
};

export default Owner;
