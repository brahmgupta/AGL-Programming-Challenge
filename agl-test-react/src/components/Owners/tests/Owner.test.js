import React from "react";
import { render } from "@testing-library/react";
import Owner from "../Owner";

const mockOwner = {
  ownerGender: "Male",
  catNames: ["Garfield", "Jim", "Max", "Tom"]
};

const mockOwnerWithNoPets = {
    ownerGender: "Male"
  };

describe("<Owner />", () => {
  it("renders the component", () => {
    const container = render(<Owner owner={mockOwner} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render All pets", () => {
    const { queryAllByTestId } = render(<Owner owner={mockOwner} />);
    expect(queryAllByTestId("pet").length).toBe(mockOwner.catNames.length);
  });

  it("should NOT render any pet", () => {
    const { queryAllByTestId } = render(<Owner owner={mockOwnerWithNoPets} />);
    expect(queryAllByTestId("pet").length).toBe(0);
  });
});
