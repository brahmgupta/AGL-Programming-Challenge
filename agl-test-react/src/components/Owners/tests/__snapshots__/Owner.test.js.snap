// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Owner /> renders the component 1`] = `undefined`;
